system/Commands/Database/ShowTableInfo.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Commands\Database;
+
+use CodeIgniter\CLI\BaseCommand;
+use CodeIgniter\CLI\CLI;
+use Config\Database;
+
+/**
+ * Get table data if it exists in the database.
+ */
+class ShowTableInfo extends BaseCommand
+{
+    /**
+     * The group the command is lumped under
+     * when listing commands.
+     *
+     * @var string
+     */
+    protected $group = 'Database';
+
+    /**
+     * The Command's name
+     *
+     * @var string
+     */
+    protected $name = 'db:table';
+
+    /**
+     * the Command's short description
+     *
+     * @var string
+     */
+    protected $description = 'Retrieves the selected table information.';
+
+    /**
+     * the Command's usage
+     *
+     * @var string
+     */
+    protected $usage = 'db:table <table_name> [options]';
+
+    /**
+     * The Command's arguments
+     *
+     * @var array<string, string>
+     */
+    protected $arguments = [
+        'table_name' => 'The table name for show information.',
+    ];
+
+    /**
+     * The Command's options
+     *
+     * @var array<string, string>
+     */
+    protected $options = [
+        '--show' => 'Retrieves list the names of all database tables.',
+    ];
+
+    /**
+     * get table info
+     */
+    public function run(array $params)
+    {
+        // Connect to database
+        $db        = Database::connect();
+        $getTables = $db->listTables();
+        // The database does not have a table
+        if ($getTables === []) {
+            return CLI::error('Database has no tables!', 'light_gray', 'red');
+        }
```

Comment:
Although `CLI::error` and `CLI::newLine` return `void`, putting them in a `return` gives the impression that they return something and, consequently, the `run` method also returns something, which is not true. So I suggest to split them.
```suggestion
        if ($tables === []) {
            CLI::error('Database has no tables!', 'light_gray', 'red');
            
            return;
        }
```

Reply:
Thank you for full explanation. It was informative, **I learned**.
In this part, a topic has occupied my mind, whether the possibility of creating a table(`createTable('table_name');`) in its absence can be useful or not? what is your opinion?

Reply:
This command is to show information. It says `Retrieves information on the selected table.`
It should not change the database state.

Reply:
> This command is to show information. It says `Retrieves information on the selected table.` It should not change the database state.

Thanks **kenjis**, I got it.